codegen/facelift/templates/Enum.template.h

```diff
@@ -74,37 +74,20 @@ template<> inline const std::initializer_list<{{enum.fullyQualifiedCppType}}>& v
     return values;
 }
 
-template <> inline QString enumToString(const {{enum.fullyQualifiedCppType}}& v)
-{
-    const char* s = "Invalid";
-    switch(v) {
-    {% for member in enum.members %}
-    case {{enum.fullyQualifiedCppType}}::{{member}}:
-        s = "{{member}}";
-        break;
-    {% endfor %}
-    default:
-        break;
-    }
-    return s;
-}
 
-}
+} // end namespace facelift
 
-
-inline void assignFromString(const QString &s, {{enum.fullyQualifiedCppType}}& v)
+inline void fromString(const QString& string, {{enum.fullyQualifiedCppType}}& value)
 {
-    {% for member in enum.members %}
-    if (s == "{{member}}")
-        v = {{enum.fullyQualifiedCppType}}::{{member}};
-    else
-    {% endfor %}
-        ::facelift::onAssignFromStringError(s);
+    auto valuePointer = facelift::Enum::fromString<{{enum.fullyQualifiedCppType}}>(string);
+    if(!valuePointer) {
+        facelift::Enum::fromStringError(string);
+    }
+    value = *valuePointer;
```

Comment:
UB?

Reply:
if there is no requested value in an enum, fromString returns nullptr inside the unique_ptr.
And is valuePointer holds nullptr, fromStringError raises qFatal.

Maybe it makes sense to move qFatal here ?

Reply:
What was the action in this/similar case previously?

Reply:
Ok, and if someone will drop qFatal in the future we don't get into UB
will add else statement.